.travis/script.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+case "$TRAVIS_OS_NAME" in
+"linux")
+    sed -i.bak "s/openjdk:8/openjdk:"${JAVA}"/" share/docker/Dockerfile
```

Comment:
This is actually equivalent to what you had before with the single quotes, because the JAVA expansion is not in quotes at all.

Instead consider 

```bash
send -i.bak "s/openjdk:8/openjdk:${JAVA}/" …
```

Bash expands names in double quotes (but treats them as a single word) and word splits unquoted expansions, which is why shellcheck would flag this line.

Reply:
Thanks, that makes sense. Just pushed an update.